pkg/cloudprovider/providers/openstack/openstack_volumes.go

```diff
@@ -92,12 +96,17 @@ func (volumes *VolumesV1) createVolume(opts VolumeCreateOpts) (string, string, e
 
 	vol, err := volumes_v1.Create(volumes.blockstorage, create_opts).Extract()
 	if err != nil {
+		recordOpenstackOperationMetric("create_v1_volume", 0, err)
 		return "", "", err
 	}
+	timeToken := time.Since(startTime).Seconds()
+	recordOpenstackOperationMetric("create_v1_volume", timeToken, nil)
```

Comment:
I think you should move all these to above the `if err != nil` handler.  ie: remove the explicitly different calls in separate branches:

```go
vol, err := volumes_v1.Create(...)
timeToken := time.Since(startTime).Seconds()
recordOpenstackOperationMetric("create_v1_volume", timeToken, err)
// rest is unchanged from original:
if err != nil {
      return "", "", err
}
return vol.ID, vol.AvailabilityZone, nil
```